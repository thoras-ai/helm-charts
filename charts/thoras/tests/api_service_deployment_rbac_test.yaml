suite: API Service Layer RBAC
templates:
  - api-server-v2/rbac.yaml
set:
  thorasVersion: TEST
  imageCredentials:
    secretRef: test-secret-registry
  thorasApiServerV2:
    rbac:
      create: true
tests:
  - it: Default matches snapshot
    chart:
      version: "4.7.0"
    asserts:
      - matchSnapshot: {}
