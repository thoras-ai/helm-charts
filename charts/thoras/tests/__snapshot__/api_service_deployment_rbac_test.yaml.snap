Default matches snapshot:
  1: |
    apiVersion: v1
    imagePullSecrets:
      - name: test-secret-registry
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: thoras
        helm.sh/chart: thoras-4.7.0
      name: thoras-api
      namespace: NAMESPACE
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: thoras
        helm.sh/chart: thoras-4.7.0
      name: thoras-api
    rules:
      - apiGroups:
          - '*'
        resources:
          - '*'
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - autoscaling
        resources:
          - horizontalpodautoscalers
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - thoras.ai
        resources:
          - '*'
        verbs:
          - '*'
      - apiGroups:
          - batch
        resources:
          - jobs
        verbs:
          - '*'
      - apiGroups:
          - apps
        resources:
          - '*'
        verbs:
          - patch
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: thoras
        helm.sh/chart: thoras-4.7.0
      name: thoras-api
      namespace: NAMESPACE
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: thoras-api
    subjects:
      - kind: ServiceAccount
        name: thoras-api
        namespace: NAMESPACE
  4: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: thoras
        helm.sh/chart: thoras-4.7.0
      name: thoras-api
      namespace: NAMESPACE
    rules:
      - apiGroups:
          - ""
        resources:
          - pods
        verbs:
          - delete
  5: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: thoras
        helm.sh/chart: thoras-4.7.0
      name: thoras-api
      namespace: NAMESPACE
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: Role
      name: thoras-api
    subjects:
      - kind: ServiceAccount
        name: thoras-api
        namespace: NAMESPACE
